Lib/test/test_capi/test_opt.py

```diff
@@ -1798,6 +1798,63 @@ def testfunc(n):
         self.assertIn("_CALL_TYPE_1", uops)
         self.assertNotIn("_GUARD_IS_NOT_NONE_POP", uops)
 
+    def test_call_tuple_1(self):
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = tuple([1, 2])  # _CALL_TUPLE_1
+                if y == (1, 2):
+                    x += 1
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertIn("_CALL_TUPLE_1", uops)
+        self.assertNotIn("_GUARD_NOS_NULL", uops)
+        self.assertNotIn("_GUARD_CALLABLE_TUPLE_1", uops)
+
+    def test_call_tuple_1_result_is_tuple(self):
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = tuple([1, 2])  # _CALL_TUPLE_1
+                if y[0] == 1:      # _BINARY_OP_SUBSCR_TUPLE_INT
+                    x += 1
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertIn("_CALL_TUPLE_1", uops)
+        self.assertIn("_BINARY_OP_SUBSCR_TUPLE_INT", uops)
+        self.assertNotIn("_GUARD_NOS_TUPLE", uops)
+
+    def test_call_tuple_1_result_propagates_for_tuple_input(self):
+        # Test a special case where the argument of tuple(arg)
+        # is known to be a tuple. The information about the
+        # argument being a tuple should be propagated to the
+        # result of tuple(arg).
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = tuple((1, 2))  # tuple argument
+                a, _ = y           # _UNPACK_SEQUENCE_TWO_TUPLE
```

Comment:
I needed to use `UNPACK_SEQUENCE(_TWO)_TUPLE` here since it is currently the only uop that propagates the tuple items to the result.

Reply:
Nice observation. Seems like `_BINARY_OP_SUBSCR_TUPLE_INT` could be taught how to do this pretty easily for a constant RHS. Want to add that to your queue?

Reply:
It's already in the queue :smile: